src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -366,122 +366,129 @@ public async Task<JObject> Resolve(ElementAccessExpressionSyntax elementAccess,
             }
         }
 
-        public async Task<JObject> Resolve(InvocationExpressionSyntax method, Dictionary<string, JObject> memberAccessValues, CancellationToken token)
+        public async Task<(JObject, string)> ResolveInvokationInfo(InvocationExpressionSyntax method, CancellationToken token)
         {
             var methodName = "";
-            bool isExtensionMethod = false;
             try
             {
                 JObject rootObject = null;
                 var expr = method.Expression;
-                if (expr is MemberAccessExpressionSyntax)
+                if (expr is MemberAccessExpressionSyntax maes)
                 {
-                    var memberAccessExpressionSyntax = expr as MemberAccessExpressionSyntax;
-                    rootObject = await Resolve(memberAccessExpressionSyntax.Expression.ToString(), token);
-                    methodName = memberAccessExpressionSyntax.Name.ToString();
+                    rootObject = await Resolve(maes.Expression.ToString(), token);
+                    methodName = maes.Name.ToString();
                 }
-                else if (expr is IdentifierNameSyntax)
-                    if (scopeCache.ObjectFields.TryGetValue("this", out JObject valueRet)) {
-                        rootObject = await GetValueFromObject(valueRet, token);
+                else if (expr is IdentifierNameSyntax && scopeCache.ObjectFields.TryGetValue("this", out JObject valueRet))
+                {
+                    rootObject = await GetValueFromObject(valueRet, token);
                     methodName = expr.ToString();
                 }
+                return (rootObject, methodName);
+            }
+            catch (Exception ex) when (ex is not ReturnAsErrorException)
+            {
+                throw new Exception($"Unable to evaluate method '{methodName}'", ex);
+            }
+        }
 
-                if (rootObject != null)
+        public async Task<JObject> Resolve(DotnetObjectId objectId, string methodName, InvocationExpressionSyntax method, Dictionary<string, JObject> memberAccessValues, CancellationToken token)
```

Comment:
Keep the signature of the method the same. And call `ResolveInvokationInfo` from here too. This would allow `MemberReferenceResolver` to not lose the ability to resolve a `InvocationExpressionSyntax`.